packages/core/components/InputOrGroup/fields/ArrayField/index.tsx

```diff
@@ -288,6 +293,10 @@ export const ArrayField = ({
 
                 <button
                   className={getClassName("addButton")}
+                  disabled={
+                    field.max !== undefined &&
+                    localState.arrayState.items.length >= field.max
+                  }
```

Comment:
Looks like we're missing disabled styles for this and the IconButton. I'll open a separate ticket for that.